src/cache/cache_las.c

```diff
@@ -1124,11 +1124,13 @@ __wt_las_sweep(WT_SESSION_IMPL *session)
 #endif
 
 		/*
-		 * If this entry isn't globally visible we cannot remove it.
-		 * If it is visible then perform additional checks to see
-		 * whether it has aged out of a live file.
+		 * If this entry isn't globally visible or if it is a part of a
+		 * prepared transaction, then we cannot remove it. Otherwise,
+		 * perform additional checks to see whether it has aged out of a
+		 * live file.
 		 */
-		if (!__wt_txn_visible_all(session, las_txnid, val_ts)) {
+		if (!__wt_txn_visible_all(session, las_txnid, val_ts) ||
+		    prepare_state == WT_PREPARE_INPROGRESS) {
```

Comment:
Why don't you need to check other prepared states here?

Reply:
@bvpvamsikrishna pointed out that updates that are in the prepared state, ie state is set to ```WT_PREPARE_INPROGRESS``` will be globally visible, but we still want the sweep to skip it. Hence I have added this check here. The state here can't be ```WT_PREPARE_LOCKED``` because that is a transient state sweep server wont see in lookaside records. The state can't be ```WT_PREPARE_RESOLVED``` either because a commit/rollback will cause the state to transition to ```WT_PREPARE_RESOLVED``` and once that happens if this record goes into the lookaside, it will be treated like a normal commited update.